### PR TITLE
GQA into Mitchich65

### PR DESCRIPTION
**Pull request by dirkgr**
TODO
 - [x] Doesn't run on AMD like this

### CODE REVIEW

File: tests/model_test.py [hunk 1]
```diff
@@ -232,6 +234,76 @@ def test_forward(
...
+        train_config.model.multi_query_attention = multi_query_attention
+        train_config.model.init_device = "cuda"
+        return Olmo(train_config.model).eval()
+
+    input1 = tokenizer.encode("As a large language model, I don’t have personal opinions.")
+    input2 = tokenizer.encode("But I can share some interesting facts!")
+    batch_inputs = DataCollator.from_train_config(train_config)(
+        [  # type: ignore
+            {"input_ids": input1, "attention_mask": [1.0] * len(input1)},
+            {"input_ids": input2, "attention_mask": [1.0] * len(input2)},
```
Comment (epwalsh) [inline tests/model_test.py:278]: I don't think we're actually using `flash-attn` here since we're passing an attention mask.

Comment (epwalsh) [reply tests/model_test.py:278]: I would just remove the mask here.

Comment (dirkgr) [reply tests/model_test.py]: Haha, yes, I noticed all those same issues and fixed them before I read your comment here 😮‍💨.

File: scripts/beaker/mitchish70.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+NUM_NODES=4
+ARGS='--device_train_microbatch_size=4'
+
+gantry run \
+  --allow-dirty \
+  --workspace ai2/llm-testing \
+  --task-name mitchish65 \
+  --description mitchish65 \
+  --priority high \
+  --beaker-image olmo-torch2-gantry \
```
Comment (dirkgr) [inline scripts/beaker/mitchish70.sh:15]: I think, at the moment, this will run on torch 2.0.

**Comment by dirkgr** [comment]
@epwalsh, the fused CE loss, will it work on LUMI?

**Comment by epwalsh** [comment]
@dirkgr I did add a config option to toggle the fused loss function in [54919e0](https://github.com/allenai/OLMo/pull/443/commits/54919e020e78d70b682b9295236b784eef2d0ed5). Defaults to false to be safe so will never run on LUMI.

File: olmo/config.py [hunk 3]
```diff
@@ -1047,6 +1049,11 @@ class TrainConfig(BaseConfig):
     The activation checkpointing strategy to use.
     """
 
+    fused_loss: Optional[bool] = None
+    """
+    Whether to use the fused CE loss function from `flash-attn`.
```
Comment (dirkgr) [inline olmo/config.py:1054]: What's the default?

File: olmo/eval/evaluator.py [hunk 4]
```diff
@@ -56,8 +56,8 @@ def compute_metrics(self) -> Dict[str, float]:
                     # get to this one within the current evaluation loop.
                     continue
                 else:
-                    out[f"eval/{label}/CrossEntropyLoss"] = loss.item()
-                    out[f"eval/{label}/Perplexity"] = torch.exp(loss).item()
+                    out[f"eval/ppl/{label}/CrossEntropyLoss"] = loss.item()
+                    out[f"eval/ppl/{label}/Perplexity"] = torch.exp(loss).item()
```
Comment (dirkgr) [inline olmo/eval/evaluator.py:60]: This will make comparisons impossible?

File: scripts/train.py [hunk 5]
```diff
@@ -241,12 +242,18 @@ def dummy_init_fn(module: torch.nn.Module) -> None:
 
 
 if __name__ == "__main__":
-    mp.set_start_method("spawn")
+    try:
+        mp.set_start_method("spawn", force=True)
+    except RuntimeError as e:
+        print(f"failed to set multiprocessing start method: {e}")
```
Comment (dirkgr) [inline scripts/train.py:248]: I did not know this could fail?

**Review by dirkgr — COMMENTED** [review]
I cannot approve this because I created the OR. I am OK with this, with two concerns in comments.

File: scripts/train.py (hunk 5 above)
Comment (epwalsh) [reply scripts/train.py:248]: Yea I saw this happen a couple times, not sure why exactly.

File: olmo/config.py (hunk 3 above)
Comment (epwalsh) [reply olmo/config.py:1054]: Defaults to `False`.
